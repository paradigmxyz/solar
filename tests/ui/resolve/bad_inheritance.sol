import "./bad_inheritance.sol" as self1;
import * as self2 from "./bad_inheritance.sol";
import {does_not_exist} from "./bad_inheritance.sol"; //~ ERROR not found in

contract C is C {} //~ ERROR contracts cannot inherit from themselves
contract D is self1.D {} //~ ERROR contracts cannot inherit from themselves
contract E is self2.E {} //~ ERROR contracts cannot inherit from themselves

contract F is self1 {} //~ ERROR expected contract, found namespace
contract G is self2 {} //~ ERROR expected contract, found namespace

contract H is does_not_exist {}
