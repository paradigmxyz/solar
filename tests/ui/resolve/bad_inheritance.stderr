error: declaration `does_not_exist` not found in $DIR/bad_inheritance.sol
  --> $DIR/bad_inheritance.sol:3:9
   |
LL | import {does_not_exist} from "./bad_inheritance.sol";
   |         ^^^^^^^^^^^^^^
   |

error: contracts cannot inherit from themselves
  --> $DIR/bad_inheritance.sol:5:15
   |
LL | contract C is C {}
   |               ^
   |

error: contracts cannot inherit from themselves
  --> $DIR/bad_inheritance.sol:6:15
   |
LL | contract D is self1.D {}
   |               ^^^^^^^
   |

error: contracts cannot inherit from themselves
  --> $DIR/bad_inheritance.sol:7:15
   |
LL | contract E is self2.E {}
   |               ^^^^^^^
   |

error: expected contract, found namespace
  --> $DIR/bad_inheritance.sol:9:15
   |
LL | contract F is self1 {}
   |               ^^^^^
   |

error: expected contract, found namespace
  --> $DIR/bad_inheritance.sol:10:15
   |
LL | contract G is self2 {}
   |               ^^^^^
   |

error: aborting due to 6 previous errors

